### PR TITLE
BUG: pandas.cut should disallow overlapping IntervalIndex bins

### LINKED ISSUE #23980: BUG: pandas.cut should disallow overlapping IntervalIndex bins
#### Code Sample, a copy-pastable example if possible

```python
In [2]: ii = pd.IntervalIndex.from_tuples([(0, 10), (2, 12), (4, 14)])

In [3]: pd.cut([5, 6], bins=ii)
Out[3]:
[(0, 10], (2, 12], (4, 14], (0, 10], (2, 12], (4, 14]]
Categories (3, interval[int64]): [(0, 10] < (2, 12] < (4, 14]]
```
#### Problem description
When an overlapping `IntervalIndex` is provided for `bins`, all intervals overlapping a given point will be returned for that point.  I don't think this behavior makes sense.

#### Expected Output
I'd expect a `ValueError` to be raised, or at the very least for this behavior to be documented if it's intended.

I think this should be straightforward to implement once `IntervalIndex.is_overlapping` has been completed, which should be soon assuming there aren't any major issues with the PR (xref #23327).

If I'm not mistaken, it should just be just a matter of modifying this block of code:
https://github.com/pandas-dev/pandas/blob/30c1290610c65d53bcdd1665be3104104ca27eb2/pandas/core/reshape/tile.py#L219-L220
And make it check if the bins are overlapping:
```python
elif isinstance(bins, IntervalIndex):
    if bins.is_overlapping:
        raise ValueError(...)    
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: 30c1290610c65d53bcdd1665be3104104ca27eb2
python: 3.6.1.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.24.0.dev0+1141.g30c129061
pytest: 3.8.2
pip: 9.0.1
setuptools: 39.0.1
Cython: 0.28.2
numpy: 1.13.3
scipy: 1.0.0
pyarrow: 0.6.0
xarray: 0.9.6
IPython: 6.1.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: 0.4.0
matplotlib: 2.0.2
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 0.9.8
lxml: 3.8.0
bs4: None
html5lib: 0.999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: 0.1.5
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [ ] closes #23980 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jschendel — CHANGES_REQUESTED** [review]
Thanks!  Looks good overall. A couple comments:
- Can you add a whatsnew note to 0.24.0 under the Reshaping subsection of the Bug Fixes section?
- Can you add "Must be non-overlapping." to `cut` docstring, at the end of the `IntervalIndex` portion of the `bins` parameter description?

**Comment by charlesdong1991** [comment]
Hi, @jschendel , thanks for the review and advices!! As for `Can you add a whatsnew note to 0.24.0 under the Reshaping subsection of the Bug Fixes section?`, I am not sure if I did it correctly (what i did was just tick the `whatsnew entry` above). Please give me more hints if it's not what you meant, sorry, not familiar with this part. And thanks again!

**Comment by jschendel** [comment]
For the whatsnew entry, you'll need to edit the `/doc/source/whatsnew/v0.24.0.rst` file at around line 1485 to add an entry along the lines of:
```rst
- Bug in :func:`cut` with ``bins`` as an overlapping ``IntervalIndex`` where multiple bins were returned per item instead of raising a ``ValueError`` (:issue:`23980`)
```

**Comment by charlesdong1991** [comment]
@jschendel , ahh, I see!! Thank you very much, added!

**Review by jschendel — APPROVED** [review]

File: pandas/core/reshape/tile.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def cut(x, bins, right=True, labels=None, retbins=False, precision=3,
           and maximum values of `x`.
         * sequence of scalars : Defines the bin edges allowing for non-uniform
           width. No extension of the range of `x` is done.
-        * IntervalIndex : Defines the exact bins to be used.
+        * IntervalIndex : Defines the exact bins to be used. Note that IntervalIndex
```
Comment (jschendel) [inline pandas/core/reshape/tile.py]: looks like linting failed since this line is a little too long

**Comment by jschendel** [comment]
thanks @charlesdong1991!

**Pull request merged**